### ISSUE TITLE
yas-snippet-mode-buffer-p doesn't detect snippets in symlinked directories

### ISSUE THREAD

**Issue by hlissner** [joaotavora/yasnippet#901]
### Steps to reproduce

+ Make a symlink of the snippets directory: `ln -s /where/i/keep/my/snippets ~/.emacs.d/snippets`
+ Open a snippet in this directory: `emacs ~/.emacs.d/snippets/text-mode/some-snippet`
+ `snippet-mode` is not active and `(yas-snippet-mode-buffer-p)` returns nil.

### Observations

`yas-snippet-mode-buffer-p` performs a basic string comparison of `buffer-file-name` and each entry  of `yas-snippet-dirs`. This will fail if a) any of these are unresolved paths or b) any entry in yas-snippet-dirs has a leading forward slash.

To address both issues I suggest this implementation:

```emacs-lisp
(defun yas-snippet-mode-buffer-p ()
  (when buffer-file-name
    (cl-loop for dir in (yas-snippet-dirs)
             if (file-in-directory-p buffer-file-name dir)
             return t)))
```

`file-in-directory-p` resolves both paths with `file-truename` and preforms a piecemeal string comparison of both paths to test equality.

**Comment by npostavs** [comment]
Oh, I somehow overlooked `file-in-directory-p`, thanks!  I've used `cl-member` instead of a loop for my fix in #902. Please test it.

**Comment by hlissner** [comment]
I've tested it and it works fine. Thanks for the quick response!
